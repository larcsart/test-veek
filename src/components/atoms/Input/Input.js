import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Input.module.scss';

const Input = ({ classes, ...props }) => (
  <input className={classNames(styles.Input, classes.root)} data-testid="Input" {...props} />
);

Input.propTypes = {
  classes: PropTypes.shape({
    root: PropTypes.string,
  }),
};

Input.defaultProps = {
  classes: {},
};

export default Input;
