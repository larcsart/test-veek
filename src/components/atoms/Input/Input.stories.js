/* eslint-disable */

import React from 'react';
import { storiesOf } from '@storybook/react';
import Input from './Input';

storiesOf('Input', module).add('default', () => {
  const inputPlaceholder = 'Text here...';
  return <Input placeholder={inputPlaceholder} />
});

