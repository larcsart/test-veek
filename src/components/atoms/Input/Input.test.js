import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Input from './Input';

describe('<Input />', () => {
  test('it should mount', () => {
    render(<Input />);

    const input = screen.getByTestId('Input');

    expect(input).toBeInTheDocument();
  });

  it('Snapshot', () => {
    const { asFragment } = render(
      <Input type="text" placeholder="Placeholder test" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
