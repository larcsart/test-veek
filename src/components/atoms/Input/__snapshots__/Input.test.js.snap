// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Snapshot 1`] = `
<DocumentFragment>
  <input
    class="Input"
    data-testid="Input"
    placeholder="Placeholder test"
    type="text"
  />
</DocumentFragment>
`;
